api/user/posts/model.py

```diff
@@ -1,39 +1,99 @@
-import enum
+from dataclasses import asdict, dataclass
 from datetime import datetime
 
-from sqlalchemy import Enum
+from sqlalchemy import Enum, event
 
 from api.app import db
+from api.user.author.model import Author
+from api.utils import Visibility, generate_object_ID, get_pagination_params
 
 
-class Visibility(enum.Enum):
-    PUBLIC = 0
-    FRIENDS = 1
-
-
+@dataclass
 class Post(db.Model):
-    id: int = db.Column(db.Integer, primary_key=True)
-    published: datetime = db.Column("published", db.DateTime, nullable=False)  # is datetime a valid way?
+    # TODO might need to change id length to accomodate other teams id
+    id: str = db.Column(db.String(50), primary_key=True, default=generate_object_ID)
+    object_id: str = db.Column(db.String(50), nullable=True)
```

Comment:
Similar to above, we can have 2 columns for easy comparison in endpoints

Reply:
also, external posts are not guaranteed to have non colliding ids. This solution will work because, the primary key for posts is the complete URL to the post which would be unique across servers. object_id field is nullable because the posts not originating on our server will not have unique ids thus keeping this field null will allow that.